cli/fathom_web/commands/histogram.py

```diff
@@ -0,0 +1,80 @@
+from json import load
+
+from click import argument, command, File, option
+from plotly.graph_objects import Histogram
+from plotly.subplots import make_subplots
+
+
+@command()
+@argument('vector_file', type=File('r'))
+@option('--feature_name', '-f',
+        default='',
+        help='The name of the feature to graph. Omit to graph all features.')
+@option('--show_zeros', '-z',
+        is_flag=True,
+        default=False,
+        help='Show zero counts in the graphs. These tend to be large and dominate the scale of the graph, making it harder to see other bars.')
+def main(vector_file, feature_name, show_zeros):
+    """Print a histogram of one or more features."""
+    data = load(vector_file)
+    feature_names = data['header']['featureNames']
+    if feature_name:
+        feature_index = feature_names.index(feature_name)
+        vectors = [yes_and_no_vectors(data['pages'], feature_index, show_zeros)]
+        feature_names = [feature_name]
+    else:
+        vectors = [yes_and_no_vectors(data['pages'], feature_index, show_zeros)
+                   for feature_index in range(len(feature_names))]
+    figure = histograms(vectors,
+                        feature_names,
+                        'Histograms of Feature Values' + ('' if show_zeros else
+                                                          ', Zeros Omitted'))
+    figure.show()
+
+
+def histograms(vectors, feature_names, title):
```

Comment:
Can we label the x- and y- axes in the plotly graphs? I imagine the x-axis is the raw feature out -- what is the y-axis? I guess I don't really know how to interpret these graphs for example:
<img width="1678" alt="Screen Shot 2020-01-08 at 8 55 37 AM" src="https://user-images.githubusercontent.com/17437436/71999045-91bc3280-31f5-11ea-9a7f-4ff6eec24fdf.png">

Also is it possible to associate a particular bar with a filename/sample name? E.g. maybe I'd like to know which negative sample has so much `textContent` in the `hasLongTextContent` feature's histogram.

Reply:
Labeling the axes is a good call. I'll see if it makes things ridiculously noisy, in which case I'll look for a central place to convey the information.

Since these are histograms, each bar is an adding-up of potentially many samples. But maybe we could put the n-heaviest in a tooltip.

Reply:
Self: All 3 of the above features in the graph should be seeing all <p> elements. Why do some see a total of hundreds and the others only 20?

Reply:
We just discussed in our 1:1 that the y-axis should be the number of candidate nodes found for each feature.

Why does the `hasEllipsisAtEndOfText` only have 20 candidate nodes (18 negative and 2 positive) while hasLongTextContent has 350+ candidate nodes? All of these rules are run without `when` clauses. Shouldn’t they all have the same number of candidate nodes?

Here's the ruleset I was using (I was only using a `vectors.json` for the paragraph ruleset):
```js
function makeRuleset(coeffs, biases) {
  return ruleset([
    /**
      * Paragraph rules
    */
    // Consider all visible paragraph-ish elements
    rule(dom("p, pre").when(isElementVisible), type("paragraph")),
    rule(dom("div").when(isElementVisible).when(divHasBrChildElement), type("paragraph").note(numParagraphTextNodesInDiv)),
    rule(dom("div").when(isElementVisible).when(divHasOnlyTextNodesAnchorElementsOrSpanElements), type("paragraph").note(numParagraphTextNodesInDiv)),
    rule(type("paragraph"), score(pElementHasListItemAncestor), {name: "pElementHasListItemAncestor"}),
    rule(type("paragraph"), score(hasLongTextContent), {name: "hasLongTextContent"}),
    rule(type("paragraph"), score(containsElipsisAtEndOfText), {name: "containsElipsisAtEndOfText"}),
    rule(type("paragraph"), score(classNameIncludesCaption), {name: "classNameIncludesCaption"}),
    // return paragraph-ish element(s) with max score
    rule(type("paragraph").max(), out("paragraph")),

    /**
      * Article rules
    */
    rule(dom("html"), type("article")),
    rule(type("article"), score(hasEnoughParagraphs), {name: "hasEnoughParagraphs"}),
    rule(type("article"), score(hasExactlyOneArticleElement), {name: "hasExactlyOneArticleElement"}),
    rule(type("article"), score(paragraphElementsHaveSiblingsWithSameTagName), {name: "paragraphElementsHaveSiblingsWithSameTagName"}),
    rule(type("article"), score(mostParagraphElementsAreHorizontallyAligned), {name: "mostParagraphElementsAreHorizontallyAligned"}),
    rule(type("article"), score(moreParagraphElementsThanListItemsOrTableRows), {name: "moreParagraphElementsThanListItemsOrTableRows"}),
    rule(type("article"), score(headerElementIsSiblingToParagraphElements), {name: "headerElementIsSiblingToParagraphElements"}),
    rule(type("article"), score(hasMultipleArticleElements), {name: "hasMultipleArticleElements"}),
    rule(type("article"), score(hasMultipleParagraphsWhoseClassNameIncludesArticle), {name: "hasMultipleParagraphsWhoseClassNameIncludesArticle"}),
    rule(type("article"), out("article"))
  ],
  coeffs,
  biases);
}
```